projects/angular-year-calendar/src/lib/pipes/week-number/week-number.pipe.ts

```diff
@@ -47,30 +72,68 @@ export class WeekNumberPipe implements PipeTransform {
     let weeksInYear;
     if (currentWeekStartDate.getTime() < firstWeekFirstDate.getTime()) {
       previousYearFirstDate = subYears(firstWeekFirstDate, 1);
-      weeksInYear = differenceInWeeks(firstWeekFirstDate, previousYearFirstDate);
+      weeksInYear = differenceInWeeks(
+        firstWeekFirstDate,
+        previousYearFirstDate
+      );
       nextYearFirstDate = firstWeekFirstDate;
       firstWeekFirstDate = previousYearFirstDate;
     } else {
       nextYearFirstDate = addYears(firstWeekFirstDate, 1);
       weeksInYear = differenceInWeeks(nextYearFirstDate, firstWeekFirstDate);
     }
 
-
     // find out the distance from the first week's first day
-    const roundFigure = ((currentWeekStartDate.getTime() - firstWeekFirstDate.getTime()) / 86400000);
-    result = (roundFigure % 7 === 0) ? roundFigure / 7 + 1 : 1 + Math.round(roundFigure / 7);
+    const roundFigure =
+      (currentWeekStartDate.getTime() - firstWeekFirstDate.getTime()) /
+      86400000;
+    result =
+      roundFigure % 7 === 0
+        ? roundFigure / 7 + 1
+        : 1 + Math.round(roundFigure / 7);
     if (result <= 0) {
       result = weeksInYear + result;
     }
 
-    const datesDiff = Math.abs(differenceInCalendarDays(nextYearFirstDate, currentWeekStartDate));
+    const datesDiff = Math.abs(
+      differenceInCalendarDays(nextYearFirstDate, currentWeekStartDate)
+    );
     if (datesDiff <= 6) {
       dateDay = Math.abs(this.getDayInView(nextYearFirstDate, weekStartsOn));
-      if (dateDay <= 3 && ((currentWeekStartDate.getMonth()) % 11) === firstWeekMonth.month) {
+      if (
+        dateDay <= 3 &&
+        currentWeekStartDate.getMonth() % 11 === firstWeekMonth.month
+      ) {
         result = 1;
       }
     }
 
+    if (ycConfig.periodWeekNumber) {
+      date.setHours(12, 0, 0, 0);
+      const diff = (date.getTime() - firstWeekFirstDate.getTime()) / 86400000;
+
+      // calculating period week number from range [1-4] using the result calculated above
+      let week = result === 53 ? 1 : result;
+      if (diff % 7 !== 0 && diff % 7 < 4) {
+        week += 1;
+      }
```

Comment:
Could you add a comment right above this if condition about what this if condition means?